### DISCUSSION TITLE
Additional instrument source formats for software synthesis; looping support

### DISCUSSION THREAD

**Discussion by peteroupc** [FluidSynth/fluidsynth#1275] [Ideas]
The following are suggestions I have to enhance FluidSynth's support for playing standard MIDI files (.mid):

- _Optional_ support for MIDI software synthesis using additional instrument source formats besides .sf2, .sf3, and .dls (e.g., Timidity and UltraSound patches; .sbk; OPL2, OPL3, and other FM instrument banks). For background, see the projects _libADLMIDI_, _libOPNMIDI_, and _OPL3BankEditor_.  SBK was also discussed in #742.
- Support for popular looping conventions seen in MIDI files, such as the RPG Maker convention.

EDIT (Oct. 9): Also, support usable seeking as described in: https://github.com/libsdl-org/SDL_mixer/issues/519

**Comment by DaforLynx** [comment]
Seconded for looping points.

**Comment by derselbst** [comment]
It is up to the application to implement looping support for MIDI files. That's not fluidsynth's job. For instance, [my player ](https://github.com/derselbst/ANMP) implements the N64 convention, while fluidsynth already provides the necessary API for doing so.

**Reply by derselbst** [reply]
Oh, and since fluidsynth is a SF2 synth, SF1 and other formats are out of scope. DLS is only supported as its synth model is widely compatible to SF2.
